.github/workflows/codechecker.yaml

```diff
@@ -0,0 +1,62 @@
+name: Run Static Analysis
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  RunCodechecker:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.9'
+
+      - name: Install Dependencies
+        run: |
+          curl -O https://efa-installer.amazonaws.com/aws-efa-installer-1.30.0.tar.gz
+          tar -xf aws-efa-installer-1.30.0.tar.gz && cd aws-efa-installer
+          sudo ./efa_installer.sh -y
```

Comment:
For a barebones install, we should only need libfabric and CUDA. Shave some runtime by just building libfabric from source.

Reply:
Some time has passed since we last discussed this, but I think that we should prefer to use the installer going forward.

I made some significant changes to the distcheck workflow and one of them is to make use of an action that uses the github action cache to accelerate apt. This should mean that we can avoid the build of libfabric entirely, if we figure out how to cache the tarball, and then just install the package.

I also like this because it would allow us to make use of [container jobs](https://docs.github.com/en/actions/using-jobs/running-jobs-in-a-container) for performing distcheck portably across AL2/AL2023/u20/u22/u24, we'd just need to take some care to call dpkg or yum depending on the target.

Reply:
If we have a binary cache for use across runs, then sure.